### DISCUSSION TITLE
SSL Connect error (Error Code:35)

### DISCUSSION THREAD

**Discussion by jetWanni** [curl/curl#13167] [Q&A]
I am getting SSL Connect error. following are the verbose logs
`LibreSSL/3.3.6: error:0DFFF0A1:lib(13):func(4095):reason(161)`

issue happens after updating mac OS to 14.2 and 14.4 ,
with 14.1 and below all 13 version had no issue

Currently my pc has curl version 8.6

what are the solutions to solve this issue,
thank you

**Comment by pkropachev** [comment]
Is curl installed from Homebrew?
Perhaps somehow related to #12525.

**Reply by jetWanni** [reply]
yes from homebrew

**Reply by jetWanni** [reply]
I followed this comment https://github.com/aws/aws-sdk-cpp/issues/2804#issuecomment-1876124188 and i tried with commenting `curl_global_cleanup` which is not a good practice but just to check if the issue is still recreating.

So my issue is solved but commenting `curl_global_cleanup` might introduce a memory leak as they said on the comment,
 So are there any permanent fix or solution for this one

**Reply by bagder** [reply]
1. #12525 is fixed in 8.6.0
2. wait, how is removing *cleanup* fixing your connect issues?
3. you're not telling us what you are doing

**Reply by jetWanni** [reply]
@bagder  sorry, i will explain here

I ve updated my pc to curl version 8.6 with homebrew and I've update the path as well  even on terminal curl -V shows 8.6  
but I've noticed that our plugin  still used  curl 8.4 version to connect (which default macOS version)

I've used to CURLVERSION_NOW to display which version plugin used to connect to the server. 

so i had to remove curl_global_cleanup from plugin source code and build again to check if its connecting.

**Reply by icing** [reply]
Sounds like you notice a bug in curl 8.4 that is fixed in 8.6. Would that summarise it?

**Reply by jetWanni** [reply]
Yes you are correct, sorry for my explanation on the main post, 
so the thing is there are customers who used our plugins and way i see we cant convince them to update curl and setup path variables, and also on macOS 14.4 its still use curl 8.4 so the way i see we have to wait till apple updating the curl right?
